### ISSUE TITLE
'Operator "-" not supported for types' but `__sub__` is typed in site-packages

### ISSUE THREAD

**Issue by Avasam** [microsoft/pyright#3876]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
I've been adding inline type comments to ImageHash so that typing is supported without separate type stubs or breaking python2 support. https://github.com/JohannesBuchner/imagehash/pull/167 Pyright doesn't see that I typed the special function `__sub__`

**To Reproduce**
This only happens when importing from site-packages. It does not happen if the offending class is declared in the same file
`pip install git+https://github.com/Avasam/imagehash.git@python2-typing#egg=ImageHash`
```python
# pyright: reportTypeCommentUsage=false
import imagehash
import numpy

hash_a = imagehash.ImageHash(numpy.array(0))
hash_b = imagehash.ImageHash(numpy.array(0))
hash_diff = hash_a - hash_b  # error
```

```python
# pyright: reportTypeCommentUsage=false
import numpy
import numpy.typing


class ImageHash(object):
    """
    Hash encapsulation. Can be used for dictionary keys and comparisons.
    """

    def __init__(self, binary_array):
        # type: (numpy.typing.NDArray[numpy.int32]) -> None
        self.hash = binary_array  # type: numpy.typing.NDArray[numpy.int32]

    def __sub__(self, other):
        # type: (ImageHash) -> int
        if other is None:
            raise TypeError('Other hash must not be None.')
        if self.hash.size != other.hash.size:
            raise TypeError('ImageHashes must be of the same shape.', self.hash.shape, other.hash.shape)
        return numpy.count_nonzero(self.hash.flatten() != other.hash.flatten())


hash_a = ImageHash(numpy.array(0))
hash_b = ImageHash(numpy.array(0))
hash_diff = hash_a - hash_b # no error
```

**Expected behavior**
Pyright should see that I can substract the class

**Screenshots or Code**
See above

**VS Code extension or command-line**
Both:
pyright 1.1.268
pylance v2022.8.41

**Comment by erictraut** [comment]
I'm not able to repro an error. Perhaps you have a stray ".pyi" file either locally or installed in your site-packages? If you right-click on ImageHash and select "Go To Declaration", where does it take you?

**Comment by Avasam** [comment]
Turns out I did have a leftover .pyi file... Sorry I was certain I had deleted it.
